### ISSUE TITLE
Crashplan Home is Kaput - other service recommendations for docker based backup clients?

### ISSUE THREAD

**Issue by johnny2678** [JrCs/docker-crashplan#46]
Curious what everyone else will be moving to.  I always found Crashplan to be slow and cumbersome on my NAS, but it did work.  Any other options?

**Comment by colinodell** [comment]
Semi-related question: will this Docker image work with CrashPlan for Small Business?

**Comment by JamRWS6** [comment]
I already had this docker image running and changed the environment variable to "PRO", upgraded my Crashplan account to the Small Business account and Crashplan sees my device and appears to be synchronizing after restarting the container.  Still waiting to see a file flow across but it appears OK.

**Comment by pconwell** [comment]
@JamRWS6 Any update? I was looking at Carbonite and Backblaze but neither seem to support linux (though I've only looked for a few minutes and haven't had time to really research the situation). Backblaze B2 is an option using rsync, but it requires some hacking together to make it work - too finicky to be a truly viable backup solution. It sucks paying twice as much for business, but if it works I'll probably just switch to crashplan business.

The other issue I'd be worried about is I _believe_ crashplan pro does not use jre/jdk and uses a native built app. So, I'd be a little concerned about them dropping support for the jre/jdk desktop app in the future and we could be back in the same boat. However, don't quote me on that, I could be wrong.

**Comment by JamRWS6** [comment]
@pconwell  It worked.  I see a file I added has been synced up to Crashplan.  Certainly a valid point how long this solution will work; probably worth looking into getting Crashplan Pro up and working in the same fashion.

The great thing is Crashplan offered me 2 free months and 75% off for 12 months for upgrading. So I'm getting business for $2.50 a month for 12 before it goes to $10.  $30 for 14 months is a great deal for me.

**Comment by johnny2678** [comment]
Do you have to backup all your files again?  Or does Crashplan Pro just
inherit your existing Crashplan Home backup?  8TB+ took almost a year to
backup - don't want to have to start over.

**Comment by pconwell** [comment]
@johnny2678 - According to the email sent out by crashplan:

> Upgrading is so easy - no need to restart your backup. In fact, you can transition your data in a few minutes.*

> \*If you’re currently using CrashPlan for Home to back up to another computer and chose to migrate to CrashPlan for Small Business, your computer-to-computer backup will be deleted once you convert. You’ll need to take action to preserve your archives prior to converting to CrashPlan for Small Business.

> Continue your backups without starting over. You can migrate your cloud backups up to 5 TB per device. Some restrictions may apply. Offer valid for current CrashPlan for Home customers as of August 1, 2017.

From their website:

> There is one exception: cloud backups that are larger than 5 TB cannot be migrated to CrashPlan for Small Business, due to technical platform constraints. 

> If one of your computers has more than 5 TB of data backed up to CrashPlan Central, that computer's cloud backup is permanently lost when you migrate. That computer starts a new backup when you migrate your account to CrashPlan for Small Business. Local backups on that same computer are not affected.

So sounds like you either need to cut your data down to below 5 TB or you are going to have to start over.

**Comment by johnny2678** [comment]
"Up to 5tb"... booo. I just renewed on Aug 17 and now they do this with no
refunds. Very bad form.

**Comment by pconwell** [comment]
@JamRWS6, this at least partially answers my concerns:

> Do I need to install a new CrashPlan app?

> No. The conversion process automatically updates the CrashPlan app on your computers. Instead of the green-themed CrashPlan for Home branding, you'll see the blue-themed CrashPlan for Small Business branding.

I have emailed tech support to ask if they are aware of any plans to move away from jre/jdk, but I'm sure I'll just get some type of canned response that doesn't actually answer my question. Either way, I'll probably go ahead and upgrade to business for now. Like you said, basically 30 bucks for a year and that's plenty of time to figure out Plan B if needed.

@johnny2678 is there any way at all to drop your backup data below 5TB? Transfer 3TB to a USB HDD? You could at least transfer _most_ of your data, then just have to upload 3TB.

**Comment by johnny2678** [comment]
Did you have to do anything to get that deal?

**Comment by pconwell** [comment]
@johnny2678, crashplan sent me an email with a link to upgrade my account. If you didn't get an email, I'd contact customer service.

**Comment by JamRWS6** [comment]
I know my home desktop version still connects to my headless synology running this container set to "PRO". I'm suspecting (at least for now) there aren't major differences between how the home and pro version interact with each other.

**Comment by pconwell** [comment]
@JamRWS6 the crashplan website has very little information about the differences between the two, but from their vague description...

> *Do I need to install a new CrashPlan app?*
> No. The conversion process automatically updates the CrashPlan app on your computers. Instead of the green-themed CrashPlan for Home branding, you'll see the blue-themed CrashPlan for Small Business branding.

Which would seem to suggest that the two apps are basically the same (or very similar). Having downloaded and installed the Small Business version, the two apps look nearly identical except the Small Business version is missing options to back up to another computer (the 'friends' option).

As far as what the option does in docker:

```bash
# Determine Crashplan Service Level to install (home or business)
if [ "$CRASHPLAN_SERVICE" = "PRO" ]; then
SVC_LEVEL="CrashPlanPRO"
else
SVC_LEVEL="CrashPlan"
 fi
```

So the only distinction is which version of the app is installed when the container is set up (in other words, the environmental variable does nothing once crashplan is installed). Now, some speculation on my part, but when I migrated my account, crashplan automatically 'upgraded' itself to the business version of the app. So, presumably, even if you installed the Home version, crashplan would 'upgrade' itself to the Small Business version once you log in with a Small Business account.

So, in summary, the `CRASHPLAN_SERVICE` environmental variable only affects what version of the app is _installed_ and once it's installed it does nothing. Additionally, crashplan _seems_ to 'upgrade' to the version according to what account type is used to sign in. Therefore it appears to be irrelevant what you set this variable to when installing crashplan docker.

**Comment by hooper303** [comment]
I didn't have to change any settings at all for the container to work with my pro account. Like @pconwell said, the login determines the app to use and it upgraded in the docker and started working on its own. I didn't try using the home win32 app with the pro backend. I was running a very old version so I figured I would upgrade anyway. After making changes to the ui_info it connected right away.

**Comment by JamRWS6** [comment]
Nice, the environment variable was the first thing I changed. Thx all for confirming this doesn't need to be done. Has anyone confirmed the version was actually auto updated after migrating their account to Pro?

**Comment by pconwell** [comment]
FYI, I emailed Crashplan tech support to ask about the Small Business app and if it would be moving away from JRE/JDK. Here is the response:

> Marie Kennedy (Code42 Helpdesk)
> Aug 30, 4:59 PM CDT
>
> Hi pconwell,
>
> Thanks for contacting Code42 Customer Support and I am happy to help you today.
>
> To my knowledge there is no plan to go native. Linus support will be supported in the jre.
>
> Do not hesitate to get back in touch if you need further support.
>
> Regards,
>
> Marie K
> Customer Support
> Code42

So take that for what it's worth, but it sounds like the app will stay JRE/JDK for the near future at least (which is a good thing for us).

@JamRWS6 My setup is a bit unique in that the crashplan service (back end) runs on a different computer and different OS than the GUI (front end) and I connect over LAN. My GUI automatically updated itself almost immediately after I migrated my account and it worked without any issues. I'm not sure if the back end would have upgraded itself or not because I manually re-installed it after I migrated my account. Presumably since the front end upgraded itself, the back end would have upgraded itself as well.

I know that's not really an answer to your question, but that's what happened with me.

**Comment by ghost** [comment]
At the moment it doesn't work for me anymore. I used this docker instance for more then a year on my QNAP NAS and last week I upgraded to the Pro version. At the same time I reformatted my QNAP and reinstalled this image from scratch. The result is that my new Crashplan Pro client on Windows (the blue one) does not talk anymore with this image fresh installed on my QNAP. Problem: password not recognized. No idea how to solve this (has nothing to do with my psw). Yes, when I install a Crashplan Pro docker from another "manufacterer" here it works fine. But I really like to use this one. Think we need another build (with a more recent back-end). I tried the environment variable setting to PRO also :)

**Comment by benfishbus** [comment]
My experience migrating was also straightforward.  Changed nothing on/in the container - just restarted it.  Installed new desktop app and re-edited .ui_info file.

The upgrade process on the CrashPlan website says that if you have >5TB, they will migrate the first 5TB and discard the rest.  I had already deleted a bunch to get under the cap, or I might have rolled the dice to see if they really cut off at 5TB.

**Comment by hooper303** [comment]
@rdebruin I am using on my QNAP just fine. So it does work. Not sure where to start as you changed so many components.

**Comment by ghost** [comment]
Think it works great if you have an existing Home edition and follow the upgrade path. I installed rungeict/crashplanpro and this works fine if you start from scratch.

**Comment by rori999** [comment]
Upgrade from home worked fine. However had some issues and needed to reinstall docker. After that no longer able to login. Its a shame this docker container worked best for me.  I'v returned to a QNAP native app which does the job, though I wish this docker container would work again.  Hopefully we get a new version of this container soon.

**Comment by vt2001cpe** [comment]
Would the outstanding changes auto update the docker container, or would we need to reinstall a container?

**Comment by AlexHello** [comment]
> Hi All,
> 
> I upgraded my home to small business restarted docker and made no changes. I cant connect from the headless UI to the docker container any more ( checked my .ui_info is pointing the right place ) and my port mapping are all setup, logging into the crashplan site says that i have not backed since i have upgraded :(
> 
> Does anyone know how to check if crashplan can be connected to from outside the docker host i think i have setup my port mapping correctly ( and made no changes since was working with home )
> 

All liess, i just forgot to update the authentication token aswell as the IP on the headless client

**Comment by pconwell** [comment]
@AlexHello, seeing as the last update to this project was 11 months ago, I'm going to say @JrCs is not actively maintaining the code. I started using [this container](https://hub.docker.com/r/jlesage/crashplan-pro/) and it works very well for crashplan small business.

EDIT: I see your question was edited and you got it working. But -- just as a note -- the container I linked does not require any special configuration of IPs or tokens, it's all through HTML5 and a web browser.
